src/TinyGsmClientXBee.h

```diff
@@ -1123,16 +1137,26 @@ class TinyGsmXBee : public TinyGsmModem<TinyGsmXBee>,
         // Put in TCP mode
         changesMade |= changeSettingIfNeeded(GF("IP"), 0x1);
       }
+      bool changesMadeCpy=changesMade;
 
       changesMade |= changeSettingIfNeeded(
           GF("DL"), String(host));  // Set the "Destination Address Low"
       changesMade |= changeSettingIfNeeded(
           GF("DE"), String(port, HEX));  // Set the destination port
 
+      if (beeType == XBEE_S6B_WIFI) {
+        DBG("modemConnect IP changes not saved");
```

Comment:
why this?

Reply:
Well its some complicated logic - and over a year since I looked through it  - so its a good question.
I think looking at the trace I made its the critical way there is not an update to the onboard persistent store. The expectation is that the non-SSL line #1138 update the IP 

```
`[2023-06-21 14:24:03.286] Sending data to [ 0 ] data.envirodiy.org
[2023-06-21 14:24:03.404] +++OK   
[2023-06-21 14:24:03.504] ATIP   
[2023-06-21 14:24:03.534] 1   
[2023-06-21 14:24:03.534] ATDL   
[2023-06-21 14:24:03.534] 0.0.0.0   
[2023-06-21 14:24:03.551] ATDL3.14.188.180   
[2023-06-21 14:24:03.566] OK   
[2023-06-21 14:24:03.566] ATDL   
[2023-06-21 14:24:03.566] 3.14.188.180   
[2023-06-21 14:24:03.598] ATDE   
[2023-06-21 14:24:03.598] 25   
[2023-06-21 14:24:03.598] ATDE50   
[2023-06-21 14:24:03.613] OK   
[2023-06-21 14:24:03.613] ATDE   
[2023-06-21 14:24:03.629] 50   
[2023-06-21 14:24:03.629] [30324] modemConnect IP changes not saved`   
```
then the POST     
then the tear down happens to ensure the link has been torn down to remote MMW   



```
[2023-06-21 14:24:05.857] ATCN   
[2023-06-21 14:24:05.857] OK   
[2023-06-21 14:24:05.973] +++OK   
[2023-06-21 14:24:06.088] ATIP   
[2023-06-21 14:24:06.104] 1   
[2023-06-21 14:24:06.104] ATDL
[2023-06-21 14:24:06.104] 3.14.188.180
[2023-06-21 14:24:06.120] ATDL127.0.0.1
[2023-06-21 14:24:06.136] OK
[2023-06-21 14:24:06.136] ATDL
[2023-06-21 14:24:06.151] 127.0.0.1
[2023-06-21 14:24:06.151] ATDE
[2023-06-21 14:24:06.168] 50
[2023-06-21 14:24:06.168] [32856] modemConnect IP changes not saved
```

I will post the Debug trace with AT listings on https://github.com/EnviroDIY/ModularSensors/issues/347  - I meant to do it - and just forgot.